pinot-core/src/main/java/org/apache/pinot/core/operator/dociditerators/SVScanDocIdIterator.java

```diff
@@ -204,6 +288,34 @@ public int matchValues(int limit, int[] docIds) {
     }
   }
 
+  private class IntMatcherAndNullHandler implements ValueMatcher {
+
+    private final ImmutableRoaringBitmap _nullBitmap;
+    private final int[] _buffer = new int[OPTIMAL_ITERATOR_BATCH_SIZE];
+
+    public IntMatcherAndNullHandler(ImmutableRoaringBitmap nullBitmap) {
+      _nullBitmap = nullBitmap;
+    }
+
+    @Override
+    public boolean doesValueMatch(int docId) {
+      if (_nullBitmap.contains(docId)) {
+        // Any comparison (equality, inequality, or membership) with null results in false (similar to Presto) even if
```

Comment:
Does this mean we should use `IS NULL` and `IS NOT NULL`? We should add it to the comment

Also, suggest either copy the comment to all the classes, or move it to the first class (`DictIdMatcherAndNullHandler`)

Reply:
For nulls to get considered, one of 4 operators can be used:

1. IS NULL
2. IS NOT NULL
3. IS DISTINCT FROM           => We need to implement it in Apache Pinot
4. IS NOT DISTINCT FROM  => We need to implement it in Apache Pinot

Please check unit tests as they demonstrate the behaviour if IS NULL or IS NOT NULL are used.